db/db_iter.cc

```diff
@@ -961,6 +960,7 @@ void DBIter::Seek(const Slice& target) {
     }
     direction_ = kForward;
     ClearSavedValue();
+    saved_key_.Reserve(saved_key_.Size() - 8);
```

Comment:
can we add more comments explaining when we are using InternalKey and when we are using UserKey and why is that